ocaml/xapi/xapi_observer.ml

```diff
@@ -599,6 +599,7 @@ let initialise_observer ~__context component =
   initialise_observer_component ~__context component
 
 let initialise ~__context =
+  Tracing.Spans.set_max_depth !Xapi_globs.max_span_depth ;
```

Comment:
do we need a similar initialization in xenopsd?
If the xapi/xenopsd propagation is fixed then the depth limit might be reached inside xenopsd instead, and currentlythat would always default to 100?